### PR TITLE
added support for inference from Model Garden

### PR DESCRIPTION
**Pull request by lkuligin**
#8850

### CODE REVIEW

**Comment by lkuligin** [comment]
@eyurtsev @hwchase17 could you take a look, please?

File: libs/langchain/langchain/llms/vertexai.py [hunk 1]
```diff
@@ -194,9 +223,41 @@ async def _acall(
...
-            self._get_task_executor().submit(self._predict, prompt, stop)
+            self._get_task_executor().submit(self._call, prompt, stop)
         )
 
+
+class VertexAIModelGarden(_VertexAIBase, LLM):
+    """Large language models served from Vertex AI Model Garden."""
+
+    client: "PredictionServiceClient" = None  #: :meta private:
+    project: str
```
Comment (baskaryan) [inline libs/langchain/langchain/llms/vertexai.py:234]: is this already defined in _VertexAIBase

File: libs/langchain/langchain/llms/vertexai.py [hunk 2]
```diff
@@ -194,9 +223,41 @@ async def _acall(
...
+    "A name of an endpoint where the model has been deployed."
+    allowed_model_args: Optional[List[str]] = None
+    """Allowed optional args to be passed to the model."""
+    prompt_arg: str = "prompt"
+    result_arg: str = "generated_text"
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that the python package exists in environment."""
+        from google.cloud.aiplatform.gapic import PredictionServiceClient
```
Comment (baskaryan) [inline libs/langchain/langchain/llms/vertexai.py:246]: the try/except ImportError should be around this line

File: libs/langchain/langchain/llms/vertexai.py [hunk 3]
```diff
@@ -214,4 +275,65 @@ def _call(
...
+
+    def _generate(
+        self,
+        prompts: List[str],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> LLMResult:
+        """Run the LLM on the given prompt and input."""
+        from google.protobuf import json_format
```
Comment (baskaryan) [inline libs/langchain/langchain/llms/vertexai.py:291]: can we wrap in try/except

**Review by baskaryan — COMMENTED** [review]
this is awesome! could we add a simple notebook to the docs showing how to use it

File: libs/langchain/langchain/llms/vertexai.py (hunk 1 above)
Comment (lkuligin) [reply libs/langchain/langchain/llms/vertexai.py:234]: It's optional in `_VertexAIBase`. But that's OK, I added a check that raises a ValueError is it's None.

File: libs/langchain/langchain/llms/vertexai.py (hunk 3 above)
Comment (lkuligin) [reply libs/langchain/langchain/llms/vertexai.py:291]: Done

**Comment by lkuligin** [comment]
> this is awesome! could we add a simple notebook to the docs showing how to use it

Added an example to the notebook.

File: libs/langchain/langchain/llms/vertexai.py (hunk 2 above)
Comment (lkuligin) [reply libs/langchain/langchain/llms/vertexai.py]: Done

**Comment by lkuligin** [comment]
@baskaryan could you take another look, please?

**Comment by baskaryan** [comment]
lgtm, thank you @lkuligin!

**Pull request merged**